### ISSUE TITLE
SigMF archive error reading data file

### ISSUE THREAD

**Issue by Deschain** [gnuradio/SigMF#244]
I think that there is an error opening sigmf archive files. The compliant dataset file is no recognized as a valid file because the relative path of the tar member is used https://github.com/gnuradio/SigMF/blob/1eef44a67636ab15078e7eecc00144b9e18cdaa3/sigmf/sigmffile.py#L779

**Comment by lgoix** [comment]
`dir = tempfile.mkdtemp()`
`dir` is created before, we could pass it to the `get_dataset_filename_from_metadata()`

**Comment by jacobagilbert** [comment]
@gmabey - you are our resident expert on Archives, can you take a look at this issue in the latest codebase?

**Comment by gmabey** [comment]
@Deschain Yes, you are correct, there is a true problem with the current implementation. Unfortunately, my best suggestion right now is to use `SigMFArchiveReader` instead, like this:
```python
import sigmf
rdr = sigmf.SigMFArchiveReader('/the/path/to.sigmf')
print(rdr.sigmffile)
```
@jacobagilbert do you think that `SigMFFile.fromfile()` should be deprecated?

**Comment by jacobagilbert** [comment]
I think we can fix this so that `fromfile()` can open all types including archives. Its used a lot.
